massa-execution-worker/src/execution.rs

```diff
@@ -677,7 +686,7 @@ impl ExecutionState {
             Ok(Response { init_cost, .. }) => {
                 self.module_cache
                     .write()
-                    .save_module(&bytecode, module, init_cost);
+                    .set_init_cost(&bytecode, init_cost);
```

Comment:
Something that I personally don't understand but that may be correct.

You set the `init_cost` only in case of success. In case of the HD cache, the module has been saved above so it's possible that the module stay save but without the init cost as they are saved in two different keys ?

Reply:
Yes that is the current behaviour, but your question reminded me that because we always compile with `max_block_gas` as input we should be able to know the init cost even after a VM error. This requires multiple changes, I will resolve the other comments.

Reply:
This has been implemented now